src/test/java/seedu/address/logic/commands/EditTaskCommandTest.java

```diff
@@ -0,0 +1,132 @@
+        expectedModel.setTask(model.getFilteredTaskList().get(0), editedTask);
+
+        assertCommandSuccess(editTaskCommand, model, expectedMessage, expectedModel);
+    }
+
+    @Test
+    public void execute_someFieldsSpecifiedUnfilteredList_success() {
+        Index indexLastTask = Index.fromOneBased(model.getFilteredTaskList().size());
+        Task lastTask = model.getFilteredTaskList().get(indexLastTask.getZeroBased());
+
+        TaskBuilder taskInList = new TaskBuilder(lastTask);
+        Task editedTask = taskInList.withName(VALID_NAME_TASK1).withDate(VALID_DATE_TASK1)
+                .withTags(VALID_TAG_MEET).build();
+
+        EditTaskDescriptor descriptor = new EditTaskDescriptorBuilder().withName(VALID_NAME_TASK1)
+                .withDate(VALID_DATE_TASK1).withTags(VALID_TAG_MEET).build();
+        EditTaskCommand editTaskCommand = new EditTaskCommand(indexLastTask, descriptor);
+
+        String expectedMessage = String.format(EditTaskCommand.MESSAGE_EDIT_TASK_SUCCESS, editedTask);
+
+        Model expectedModel = new ModelManager(new AddressBook(model.getAddressBook()),
+                model.getTaskBook(), new StrategyBoard(), new UserPrefs());
+        expectedModel.setTask(lastTask, editedTask);
+
+        assertCommandSuccess(editTaskCommand, model, expectedMessage, expectedModel);
+    }
+
+    @Test
+    public void execute_noFieldSpecifiedUnfilteredList_success() {
+        EditTaskCommand editTaskCommand = new EditTaskCommand(TASK_FIRST_INDEX, new EditTaskDescriptor());
+        Task editedTask = model.getFilteredTaskList().get(TASK_FIRST_INDEX.getZeroBased());
+
+        String expectedMessage = String.format(EditTaskCommand.MESSAGE_EDIT_TASK_SUCCESS, editedTask);
+
+        Model expectedModel = new ModelManager(new AddressBook(model.getAddressBook()),
+                model.getTaskBook(), new StrategyBoard(), new UserPrefs());
+
+        assertCommandSuccess(editTaskCommand, model, expectedMessage, expectedModel);
+    }
+
+    @Test
+    public void execute_duplicatePersonUnfilteredList_failure() {
+        Task firstTask = model.getFilteredTaskList().get(TASK_FIRST_INDEX.getZeroBased());
+        EditTaskDescriptor descriptor = new EditTaskDescriptorBuilder(firstTask).build();
+        EditTaskCommand editTaskCommand = new EditTaskCommand(TASK_SECOND_INDEX, descriptor);
+
+        assertCommandFailure(editTaskCommand, model, EditTaskCommand.MESSAGE_DUPLICATE_TASK);
+    }
+
+    @Test
+    public void execute_invalidPersonIndexUnfilteredList_failure() {
+        Index outOfBoundIndex = Index.fromOneBased(model.getFilteredTaskList().size() + 1);
+        EditTaskDescriptor descriptor = new EditTaskDescriptorBuilder().withName(VALID_NAME_TASK1).build();
+        EditTaskCommand editTaskCommand = new EditTaskCommand(outOfBoundIndex, descriptor);
+
+        assertCommandFailure(editTaskCommand, model, Messages.MESSAGE_INVALID_TASK_DISPLAYED_INDEX);
+    }
+
```

Comment:
You might want to add additional test cases to ensure that the persons check is working properly. You can refer to the test cases for AddTaskCommand